<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="apple-mobile-web-app-capable" content="yes">

<style>
@import url(CSS/stack.css);
@import url(CSS/d3-tip.css);
@import url(CSS/figures.css);
@import url(CSS/kbroman_talk.css);
@import url(CSS/corr_w_scatter.css);
</style>

<script charset="utf-8" type="text/javascript" src="JS/d3.min.js"></script>
<script type="text/javascript" src="JS/d3.tip.js"></script>
<script type="text/javascript" src="JS/slide_transition.js"></script>


<title>Opportunities and challenges in the genetic analysis of
high-throughput phenotypes</title>
</head>
<body class="overview">



<section class="titlepage">
<h3 style="margin-top:-10px;" class="highlight">Opportunities and
challenges<br/>
in the genetic analysis of<br/>
high-throughput phenotypes</h2>

<h4><a href="http://kbroman.org">Karl Broman</a></h4>
<p class="nopadding">Biostatistics &amp; Medical Informatics, University of Wisconsin
&ndash; Madison</p>

<p class="nopadding"><code>
<a href="http://kbroman.org" class="gray">kbroman.org</a><br/>
<a href="https://github.com/kbroman" class="gray">github.com/kbroman</a><br/>
<a href="https://twitter.com/kwbroman" class="gray">@kwbroman</a>
</code></p>

<p class="nopadding"><span class="gray">slides</span>: <code><a href="http://www.biostat.wisc.edu/~kbroman/presentations/Texas2015">bit.ly/Texas2015</a></code></p>

<img style="position:fixed;bottom: 30px; right:40px; z-index: -1;" width="150px" src="Figs/cc-zero.png" />

<aside>
These are slides for a talk I will give for a talk at
Texas A&amp;M on 1 Sep 2015.<br/><br/>

Click on the link at the top-right to remove these
notes.<br/><br/>

These slides were tested for Chrome on a Mac and might look terrible
in some browsers.
</aside>

</section>





<section>
<h3>Inbred mice</h3>

<img style="position:fixed;bottom: 0px; left:0px; z-index: -1;" width="1280px" src="Figs/inbredmice.jpg" />

<aside>
I focus on genetics problems, and particularly on mouse genetics.<br/><br/>

I think these are <a
href="http://jaxmice.jax.org/strain/000689.html">SWR mice</a>; the
photo is from <a
href="http://cals.ncsu.edu/genetics/index.php/people/david-threadgill/">David
Threadgill</a>.
</aside>

<p class="pg">2</p></section>






<section>
<h3>Human vs mouse</h3>

<img style="position:fixed;top: 300px; left:120px; z-index: -1;" height="500px" src="Figs/da-vinci-man.jpg" />

<img style="position:fixed;top: 300px; right:120px; z-index: -1;" height="500px" src="Figs/vitruvian_mouse.jpg" />

<aside>
Mice are not humans, but you can learn a great deal about human
biology and disease from mice.<br/><br/>

The figure on the right is from <a
href="http://www.daviddeen.com">David Deen</a>.
</aside>
<p class="pg">3</p></section>






<section>
<h3>Genome scan for QTL</h3>

<img src="Figs/lodcurve_insulin.png" />

<aside>
I&rsquo;ve focused largely on the analysis of a single phenotype. The
simplest analysis is to scan the genome, assessing the association
between genotype and phenotype, considering each position
individually.<br/><br/>

I&rsquo;ve focused particularly on multi-locus methods: to identify the set
of loci that are well supported, including
epistatic interactions and the possibility of linked QTL.<br/><br/>

But in many cases the single-locus scan works remarkably well and is
hard to beat.
</aside>
<p class="pg">4</p></section>





<section>
<h3>Genome scan for QTL</h3>

<img src="Figs/lodcurve_insulin_with_effects.png" />

<aside>
Underneath each of these peaks is an association between genotype and
phenotype: If you group the mice by genotype at that location, you&rsquo;ll
see a difference in the average phenotype among the groups.<br/><br/>

The BB, BR, and RR indicate the three different genotypes, as these
data concern a cross between two strains labeled B and R.
</aside>
<p class="pg">4</p></section>






<section>
<h3>Genome-scale phenotypes</h3>

<img height=700 class="indent" src="Figs/mouse_on_chips.png" />

<aside>
This is a picture of a pile of gene expression arrays. More and more,
we&rsquo;re seeing genome-scale phenotype information. For example, in
one of my collaborations, we have data on 500 mice, each with gene
expression microarrays for 6 different tissues.
We also have numerous &ldquo;clinical&rdquo; phenotypes.<br/><br/>

The move is clearly towards deeper and more extensive
phenotyping: in model organisms (as I study), in humans (particularly
focusing on electronic medical records), and in plants
(in the greenhouse or in the field).<br/><br/>

Particularly with experimental organisms, the high-dimensional
phenotypes are often measured in multiple environments, with multiple
treatments, and over time.
</aside>
<p class="pg">5</p></section>




<section>
<div class="center"><h3>Why?</h3></div>

<aside>
People embark on high-throughput phenotyping projects for a variety of
different reasons.<br/><br/>

They may just want to learn more things, or work together with others,
sharing the costs of gathering and genotyping individuals and
developing a coordinated phenotyping protocol. And one may identify
important relationships among traits.<br/><br/>

In mouse genetics, we&rsquo;re particularly interested in speeding the
effort to identify genes underlying complex clinical traits, and to
improve our understanding of the underlying disease etiology.<br/><br/>

In many cases, the high-throughput phenotyping is a potentially
less-expensive proxy for the real trait of interest. This is
particularly the case for electronic medical records (EMRs), which
often seem more related to medical billing than to medical
research.<br/><br/>

Or we may just have no idea what to measure (or in what tissue to
measure it), and so we try everything, just as we measure genotype
genome-wide rather than at specific candidate genes.<br/><br/>

In agriculture (and to some extent with humans), the primary interest
may be in prediction the future course of disease, or to get an early
understanding of a trait that can&rsquo;t be measured until much
later. (For example, in plants, early growth as a predictor of
final yield.)
</aside>
<p class="pg">6</p></section>


<section>
<div class="center"><h3>But more is not necessarily better</h3></div>

<aside>
There is often the feeling that if we measure more things, we&rsquo;ll
be better off.<br/><br/>

But if our high-throughput phenotypes have considerably higher
measurement variation, we may have a harder time identifying
genes.<br/><br/>

I think ideally we want to measure lower-level traits that are closer
to the genes and have simpler genetic architecture. (That is, are
influenced by fewer QTL, and with stronger QTL effects.)
</aside>
<p class="pg">7</p></section>




<section>
<h3>Challenges: <span class="lowlight">diagnostics</span></h3>

<img height=600 class="center" src="Figs/weird_correlation_matrix.png"/>

<p class="codelink">
<a href="https://kbroman.wordpress.com/2012/04/25/microarrays-suck">kbroman.wordpress.com/2012/04/25/microarrays-suck</a>
</p>
<aside>
This is a correlation matrix for a set of arrays. What the heck is that?
</aside>
<p class="pg">8</p></section>


<section>
<h3>Challenges: <span class="lowlight">diagnostics</span></h3>

<img width=900 class="center" src="Figs/many_boxplots.png"/>

<p class="codelink">
<a href="https://kbroman.wordpress.com/2012/04/25/microarrays-suck">kbroman.wordpress.com/2012/04/25/microarrays-suck</a>
</p>
<aside>
This is like a set of 487 boxplots of the array data, sorted by their
median. The black line is at the median. The pink lines are at the
25th and 75th percentiles. The green, orange, and blue lines are at 10,
5, and 1 percentiles.<br/><br/>

It turned out that there were a batch of 120 badly-behaved arrays.<br/><br/>

Things to be concerned about: sample mix-ups, batch effects, bad
assays.<br/><br/>

But it&rsquo;s hard to look at the data, given the scale.
</aside>
<p class="pg">9</p></section>



<section>
<h3>Challenges: <span class="lowlight">metadata</span></h3>

<div class="center"><h4>What the heck is &ldquo;<code class="highlight">FAD_NAD SI 8.3_3.3G</code>&rdquo;?</h4></div>

<aside>
An actual phenotype name.<br/><br/>

We need to create data dictionaries with short and long versions of
names and then proper descriptions. It can be helpful to categorize
the phenotypes, as well.
</aside>
<p class="pg">10</p></section>


<section>
<h3>Challenges: <span class="lowlight">Organizing & automating</span></h3>

<div class="center" id="orgchart"></div>


<aside>
The phenotypes (and samples) typically come in batches. Each batch
will need a bit of curation (e.g., diagnostics).<br/><br/>

Our ideas of what analyses to perform generally evolve over the course
of a project. With a single phenotype, this is pretty easy: you just
revise your scripts and re-run.<br/><br/>

With <span class="highlight">many</span> batches of <span
class="highlight">many</span> phenotypes, you may need to go back and
re-do things. How to keep track of what was done for which
phenotypes?<br/><br/>

I suggest having a table that describes, for each phenotype or batch
of phenotype, what needs to be done. And the results should include
some record of how they were obtained. And then you run some
uber-script that checks to see what needs to be done or re-done
and ships off the computations and collects the results.<br/><br/>

<span class="highlight">Animations</span>: click on the
&ldquo;genotypes&rdquo; box to have more batches of individuals
displayed. Then click on the &ldquo;phenotypes&rdquo; box to have more
batches of phenotypes displayed.
</aside>
<p class="pg">11</p></section>


<section>
<h3>Challenges: <span class="lowlight">Scale of the results</span></h3>


<div class="center" id="scale_of_results"></div>

<aside>
You think the data are big; the results are <span
class="highlight">really</span> big, even if you just do pairwise
associations (phenotype vs phenotype; phenotype vs
genotype).<br/><br/>

<span class="highlight">Animation</span>: click on the
either of the two boxes to have the &ldquo;results&rdquo; box
displayed. The point is that it's much bigger.
</aside>
<p class="pg">12</p></section>



<section>


<div class="center"><h3 class="lowlight">What was the question again?</h3></div>


<aside>
It&rsquo;s easy to lose track of what you&rsquo;re supposed to be
doing, particularly given the scale of the results.
</aside>
<p class="pg">13</p></section>





<section>
<h3>The ridiculome</h3>

<img height=600 class="center" src="Figs/hairball_kim_pnas_2007_104_20274.png" />



<p class="codelink">
<a href="http://www.pnas.org/content/104/51/20274">Kim et al (2007) PNAS 104:20274&ndash;20279</a>
</p>

<aside>
Many questions about high-throughput phenotypes end up concerning the
network of relationship among traits and loci.<br/><br/>

Too often, efforts on this end up with this sort of hairball network
that doesn&rsquo;t really teach us anything.
</aside>
<p class="pg">14</p></section>





<section>
<h3>Pleiotropy?</h3>

<img class="center" height=600 src="Figs/pleiotropy_network.svg"/>

<aside>
A common question is whether phenotypes are influenced by a common QTL
or two closely linked QTL.<br/><br/>

You can think of this as concerning the structure of a network.
</aside>
<p class="pg">15</p></section>



<section>
<h3>Causal?</h3>

<img class="center" height=600 src="Figs/causal_network.svg"/>

<aside>
We're also interested in whether some low-level trait sits between the
QTL and the trait, and might actually <span
class="highlight">be</span> the QTL.<br/><br/>

Is it a cause or a symptom?
<br/><br/>

You can think of this as concerning the structure of a network.
</aside>
<p class="pg">16</p></section>


<section>
<h3>Multivariate analysis</h3>

<img height=600 class="center" src="Figs/multivariate.png" />


<aside>
The joint analysis of multiple phenotypes can be revealing of
underlying QTL effects.<br/><br/>

These are scatterplots of the expression of two pairs of genes in 500
intercross mice, with the points colored according to mouse genotype
at a QTL.<br/><br/>

(So really bivariate analysis, not multivariate.)
<br/><br/>

The plot on the right is particularly interesting: conditional on QTL
genotype, the expression of the two genes is positively correlated,
but one is affected positively by the QTL and one negatively.
</aside>
<p class="pg">17</p></section>


<section>
<h3>Multivariate analysis</h3>

<img height=600 class="center" src="Figs/multivariate2.png" />


<aside>
This is another pair of genes; blue, green and red are genotypes at a
QTL, for mice that are <span class="highlight">not recombinant</span>
across the region. The yellow points are recombinant mice.<br/><br/>

That the recombinant mice show different phenotype pattern than the
non-recombinants indicates that there are two linked QTL.
</aside>
<p class="pg">18</p></section>






<section>
<h3>Composite phenotypes</h3>

<img height=600 class="center" src="Figs/feingold_dentalcaries.png" />



<p class="codelink">
<a href="http://jdr.sagepub.com/content/92/1/32">Shaffer et al. (2013) J Dent Res 92:32-37</a>
</p>

<aside>
This is my favorite example of the construction of composite traits:
the clustering of tooth surfaces by the co-occurrence of dental caries,
by <a
href="http://www.publichealth.pitt.edu/home/directory/john-r-shaffer">John
Shaffer</a> and colleagues at the University of Pittsburgh.<br/><br/>

The different colors correspond to the inferred clusters, determined
without consideration of spatial positioning.<br/><br/>

The investigators were not able to identify any susceptibility loci
when considering individual surfaces, but did when they focused on
these inferred clusters.
</aside>
<p class="pg">19</p></section>



<section>

<div class="center"><h3>Share more data, sooner.</h3></div>

<aside>
Having spent a ton of money on phenotyping, and been unable to fully
explore it, one may be tempted to hold it back for further analysis.<br/><br/>

But I think one should just admit that there is more to be found, and
that we will take best advantage of our investment in the phenotyping
by putting multiple data sets together and having many people
continuing to study valuable data.<br/><br/>

We need to do a better job of giving credit to people who have
developed and distributed important data sets.
</aside>
<p class="pg">20</p></section>


<section>
<h3>Training</h3>

<ul class="tight">

  <li>Data analysis and visualization</li>

      <ul class="lowlight">
        <li><a href="https://stat545-ubc.github.io">Stat 545 at
        UBC</a></li>
        <li><a
        href="https://www.coursera.org/specialization/jhudatascience/1">Johns
        Hopkins/Coursera data science series</a></li>
      </ul>

  <li>Project and data organization</li>

      <ul class="lowlight">
        <li><a href="http://datacarpentry.org">Data Carpentry</a></li>
      </ul>

  <li>Software engineering</li>

      <ul class="lowlight">
        <li><a href="http://software-carpentry.org">Software Carpentry</a></li>
      </ul>
</ul>

<aside>
To be able to manage and study high-throughput phenotyping, biologists
and statisticians and others are going to need to improve their
practices, from visualization to programming.<br/><br/>

Statistics courses often focus on small-scale, nicely-behaved
data. The curriculum needs to be modernized for our modern needs. <a
href="http://www.stat.ubc.ca/~jenny/">Jenny Bryan</a>&rsquo;s <a
href="https://stat545-ubc.github.io">Stat 545</a> course at the
University of British Columbia is a really good example. Also look at
the data science Coursera courses from <a
href="http://www.bcaffo.com/">Brian Caffo</a>, <a
href="http://jtleek.com">Jeff Leek</a>, and <a
href="http://www.biostat.jhsph.edu/~rpeng/">Roger Peng</a>, at Johns
Hopkins.<br/><br/>

Even more, we all need to learn to better manage our large data
projects. <a href="http://datacarpentry.org">Data Carpentry</a> is an
important effort along these lines.<br/><br/>

And we need to improve not just our programming skills but really our
software engineering skills: program organization, documentation,
version control, and testing.  <a
href="http://software-carpentry.org">Software Carpentry</a> provides
provides important courses on these topics, for scientists.
</aside>
<p class="pg">21</p></section>




<section>
<h3>Interactive graphics</h3>

<div id="corr_w_scatter" style="margin-left:-50px;"></div>

<script type="text/javascript" src="JS/corr_w_scatter.js"></script>

<aside>
A major part of my effort to conquer the high-throughput phenotyping
problems is to develop interactive data visualization that enable a
deeper and more rapid exploration of the data and analysis results.<br/><br/>

Heatmaps of things like correlation matrices can be quite useful, but
they are often hard to read.  It&rsquo;s nice to be able to read off the
values by hovering.<br/><br/>

Even better: to be able to click on a pixel and view the corresponding
scatterplot.<br/><br/>

This is a set of genes whose expression values are associated with
genotype at a common locus. In the scatterplot, the points are colored
by genotype at that locus.<br/><br/>

With high-dimensional data, data visualizations will necessarily
involve compressed summaries. But with interactive data
visualizations, one can have immediate access to the underlying details.
</aside>

<p class="codelink">
<a href="Coffee/corr_w_scatter.coffee">code</a> |
<a href="Data/corr_w_scatter.json">data</a></p>
<p class="pg">22</p></section>



<section>
<h3 class="smaller">All graphs can be improved with interactivity</h3>

<div id="lod_and_effect" class="fig"></div>

<aside>

While my efforts towards interactive graphs were motivated by the
needs for high-dimensional data, I&rsquo;ve come to the conclusion that
basically all graphs could be improved with at least a bit of
interactivity. You at least want tooltips.<br/><br/>

We know we should not rely solely on summary statistics (like the LOD
score), but if it takes an extra step to plot the underlying details,
we tend not to do it, or at least not so much as we should.<br/><br/>

The curves at the top are the same LOD curves we saw before. The LOD
score is a measure of association between genotype and phenotype.<br/><br/>

Click on a chromosome in the top panel to see a detailed view in the
lower-left panel.  Click on a marker in the lower-left panel to see
effect estimates and raw data on the right.  Hover over things on the
right to get some further information.
</aside>

<p class="codelink">
<a href="Coffee/lod_and_effect.coffee">code</a> |
<a href="Data/insulinlod.json">data</a></p>

<script type="text/javascript" src="JS/lod_and_effect.js"></script>
<p class="pg">23</p></section>




<section>
<h3 class="smaller">Genome scan for a longitudinal phenotype</h3>

<div id="lod_alltimes_fig" style="font-size:20px;margin-left:-50px;"></div>

<aside>
Here we consider a phenotype measured over time, and perform QTL
analysis with each time point individually, to seek genetic loci that
affect the response.<br/><br/>

It&rsquo;s a lot of work to make sense of these results without an
interactive graph.<br/><br/>

The top-left plot is a &ldquo;heat map&rdquo; of the LOD scores
for each time point at each genomic position.  LOD scores are also
colored to indicate the sign of the QTL effect, with red indicating
that BB lines have larger phenotype values and blue indicating that AA lines
have larger phenotype values.  We consider only those (position, time)
pairs with LOD > 1.<br/><br/>

When you hover over a point in the top-left plot, the LOD curves for the
corresponding time are shown below, and the phenotype averages and
estimated QTL effect (across time) for the corresponding genomic
position are shown to the right.<br/><br/>

Click on a point to show pointwise confidence bands on the QTL
effect (&plusmn; 2 SE).  (We require a mouse click, as otherwise the graph
was painfully slow to refresh.)
</aside>


<p class="codelink">
<a href="Coffee/lod_alltimes.coffee">code</a> |
<a href="Data/all_lod.json">data</a>

<script type="text/javascript" src="JS/lod_alltimes.js"></script>


</aside>

<!--
<p class="codelink">
<a href="Coffee/corr_w_scatter.coffee">code</a> |
<a href="Data/corr_w_scatter.json">data</a></p>
-->
<p class="pg">24</p></section>



<section>
<h3>Interactivity is great for teaching</h3>

<div id="lod_onetime_random_fig" style="margin-left:-30px;"></div>

<p class="codelink">
<a href="Coffee/lod_onetime_random.coffee">code</a> |
<a href="Data/onetime_random.json">data</a>


<script type="text/javascript" src="JS/lod_onetime_random.js"></script>

<aside>
Interactive graphs are incredibly valuable educational tools.<br/><br/>

You can only show so many static graphs, and it can be a big step to get
students to make more, on their own.<br/><br/>

Interactive graphs can be a big help to build students&rsquo; intuition
and understanding, both as a demonstration, and as a tool that they
can play with.<br/><br/>

This is an illustration of permutation test: Click the green button to shuffle
the phenotypes and re-draw the LOD curves.<br/><br/>

There&rsquo;s a hidden &ldquo;back&rdquo; button a bit to the right of
the green button.
</aside>

<p class="pg">25</p></section>




<section nopadding>
<h3 style="margin-top:10px;margin-left:-20px;">Interactive eQTL plot</h3>

<div id="cistrans" class="fig" style="margin-top:-80px;"></div>

<aside>
Here&rsquo;s my most fancy plot. But when I showed this to my
collaborator, his first question was, &ldquo;Where&rsquo;s my favorite
gene?&rdquo; I needed to add a search box. <em>Moral</em>: These tools
are not just for playing around; they need to address the scientific
questions.<br/><br/>

The top-left image shows inferred eQTL (sites in genome where genotype
is associated with the expression of some gene).  The y-axis
corresponds to the genomic location of probes on a gene expression
array.  The x-axis corresponds to marker positions.  Dots indicate
that the genotypes at a particular marker are associated with the
expression of a particular probe.<br/><br/>

Hover over a point in the top-right to see all eQTL for a given probe.
Click on a point to see the LOD curves across the genome for that
probe: the association between that probe&rsquo;s expression and genotype at
each position in the genome.<br/><br/>

Click on a marker in the lower LOD curve plot to see the corresponding
phenotype-vs-genotype plot in the top-right.
</aside>

<p class="codelink">
<a href="Coffee/cistrans.coffee">code</a></p>

<script type="text/javascript" src="JS/cistrans.js"></script>
<p class="pg">26</p></section>





<section nopadding>
<h3><a href="http://d3js.org">D3</a> is awesome!</h3>

<div id="learnd3" class="medtop">
<p>You <span class="darkgray">just</span> need to learn html, css, svg, and javascript.</p>
<p id="dotenter" style="opacity:0;">And don&rsquo;t forget <code>.enter()</code></p>
</div>

<div id="collision" style="position:fixed;top:-10px;left:-50px;width:1280px;height:960px;">
</div>

<div style="position:fixed;bottom:10px;left:10px;">
<code style="font-size:28px;"><a href="http://mbostock.github.io/d3/talk/20111018/collision.html">http://mbostock.github.io/d3/talk/20111018/collision.html</a></code></div>

<aside>
I&rsquo;m using <a href="http://d3js.org">D3</a>, a javascript
library for data visualization.  (D3 stands for &ldquo;data-driven
documents.&rdquo;)<br/><br/>

This is one of the images that convinced me that I wanted to learn D3.
There are lots of additional examples at the <a
href="https://github.com/mbostock/d3/wiki/Gallery">D3
gallery</a>.<br/><br/>

There&rsquo;s a ton to learn, and it&rsquo;s rather forbidding (and much of it is
dull).<br/><br/>

My most common mistake is to forget to include <code>.enter()</code>.
No error is given, but nothing shows up in the plot.
</aside>
<p class="pg">27</p></section>
<script type="text/javascript" src="JS/collision.js"></script>



<section>
<h3><a class="white" href="http://kbroman.org/qtlcharts">R/qtlcharts</a></h3>


<table cellpadding=25 style="margin-top:50px;">
  <tr><td valign="top"><a href="http://kbroman.org/qtlcharts/example/iplotScanone.html">
               <img src="Figs/iplotScanone.png" title="iplotScanone" width=325/></a></td>
      <td valign="top"><a href="http://kbroman.org/qtlcharts/example/iplotCorr.html">
               <img src="Figs/iplotCorr.png" title="iplotCorr" width=325/></a></td>
      <td valign="top"><a href="http://kbroman.org/qtlcharts/example/iplotMScanone.html">
               <img src="Figs/iplotMScanone.png" title="iplotMScanone" width=325/></a></td>
  </tr>
  <tr><td valign="top"><a href="http://kbroman.org/qtlcharts/example/iplotCurves.html">
               <img src="Figs/iplotCurves.png" title="iplotCurves" width=325/></a></td>
      <td valign="top"><a href="http://kbroman.org/qtlcharts/example/iheatmap.html">
               <img src="Figs/iheatmap.png" title="iheatmap" width=325/></a></td>
      <td valign="top"><a href="http://kbroman.org/qtlcharts/example/iboxplot.html">
               <img src="Figs/iboxplot.png" title="iboxplot" width=325/></a></td>
  </tr>
</table>


<aside>
I want to be able to generate interactive graphs rapidly: for
my own exploratory data analysis, and for my many collaborators.<br/><br/>

My tendency is to write a script with a bunch of code for a specific chart
with specific data.  I need to fight that and write reusable functions.<br/><br/>

And I need a system for generating interactive charts, so that others
(including myself a few months from now) can make use of what I&rsquo;ve
built.<br/><br/>

I&rsquo;m writing an R package, <a
href="http://kbroman.org/qtlcharts">R/qtlcharts</a> for
interactive graphs for QTL mapping data, as a companion to my <a
href="http://www.rqtl.org">R/qtl</a> package. It writes html
(including the data, in <a href="http://www.json.org/">JSON</a>
format, and links to the javascript code) to a file and then opens it
in a browser. For small-to-medium-scale charts, this is fast and
effective.
</aside>
<p class="pg">28</p></section>




<section>
<h3 style="margin-top:10px">Summary</h3>

<ul class="tight" style="margin-top:-10px">

  <li>Challenges</li>

      <ul class="lowlight">
        <li>Organizing analyses</li>
        <li>Making sense of the enormous pile of results</li>
        <li>Inferring networks; breaking down to small networks</li>
        <li>Joint consideration of multiple traits; form composite traits</li>
      </ul>

  <li>Share more data, sooner</li>

  <li>Needed upgrades in training</li>

      <ul class="lowlight">
        <li>Data analysis and visualization</li>
        <li>Data management</li>
        <li>Software engineering</li>
      </ul>

  <li><span class="highlight">Interactive</span> graphics
  require effort, but they</li>

      <ul class="lowlight">
        <li>Facilitate exploration</li>
        <li>Are great collaborative tools</li>
        <li>Enable summaries with access to the details</li>
        <li>Are great for teaching</li>
        <li>Need to be <span class="highlight">tailored</span> to the data and questions</li>
       </ul>

  <li><a href="http://kbroman.org/qtlcharts" class="highlight">R/qtlcharts</a>,
        <a class="gray" href="http://www.ncbi.nlm.nih.gov/pubmed/25527287">Broman (2015) Genetics 199:359&ndash;361</a></li>

</ul>

<aside>
It&rsquo;s always good to have a summary slide with your key points.
</aside>
<p class="pg">29</p></section>





<section>
<h3>Acknowledgments<span class="lowlight"></span></h3>

<div style="margin-top:-20px;margin-left:120px;font-size:28px;">

<table cellspacing=25>
  <tr valign="top">
     <td class="lowlight">Alan Attie<br/>Mark Keller</td>
     <td width=10></td>
     <td>Biochemistry, UW&ndash;Madison</td>
     </tr>

  <tr valign="top">
     <td class="lowlight">Brian Yandell</td>
     <td></td>
     <td>Statistics and Horticulture, UW&ndash;Madison</td>
     </tr>

  <tr valign="top">
     <td class="lowlight">Christina Kendziorski<br/>Aimee Teo Broman</td>
     <td></td>
     <td>Biostatistics & Medical Informatics, UW&ndash;Madison</td>
     </tr>

  <tr valign="top">
     <td class="lowlight">Eric Schadt
     <td></td>
     <td>Mount Sinai</td>
     </tr>

  <tr valign="top">
     <td class="lowlight">Danielle Greenawalt<br/>Amit Kulkarni</td>
     <td></td>
     <td>Merck & Co., Inc.</td>
     </tr>

  <tr height=25><td></td></tr>

  <tr valign="top">
     <td class="lowlight">&#346;aunak Sen</td>
     <td></td>
     <td>Epidemiology & Biostatistics, UC-San Francisco</td>
     </tr>
  <tr height=25><td></td></tr>

  <tr valign="top">
     <td class="lowlight">Edgar Spalding<br/>Candace Moore<br/>Logan Johnson</td>
     <td></td>
     <td>Botany, UW-Madison</td>
     </tr>
</table>
</div>

<aside>
Many people are involved in the studies that I presented.
</aside>
<p class="pg">30</p></section>



<section>
<div class="hugetop">
<p class="large"><span class="gray">slides</span>: <code><a href="http://www.biostat.wisc.edu/~kbroman/presentations/Texas2015">bit.ly/Texas2015</a></code>
&nbsp; <img style="vertical-align:middle;" width="150px" src="Figs/cc-zero.png" /></p>

<p class="large"><code>
<a href="http://kbroman.org" class="gray">kbroman.org</a><br/>
</code></p>

<p class="large"><code>
<a href="https://github.com/kbroman" class="gray">github.com/kbroman</a><br/>
</code></p>

<p class="large"><code>
<a href="https://twitter.com/kwbroman" class="gray">@kwbroman</a>
</code></p>

<p class="large"><code>
<a href="http://kbroman.org/qtlcharts" class="gray">kbroman.org/qtlcharts</a>
</code></p>

</div>

<aside>
This is where to find me.<br/><br/>

In particular, you can find all of my papers, talks, and software.
</aside>
<p class="pg">31</p></section>



<script type="text/javascript" src="JS/stack.v0.js"></script>
<script type="text/javascript" src="JS/handle_stickies.js"></script>
<script type="text/javascript" src="JS/org_automate.js"></script>
<script type="text/javascript" src="JS/scale_of_results.js"></script>
</body>
</html>
